src/corehost/cli/apphost/bundle/manifest.h

```diff
@@ -5,44 +5,44 @@
 #ifndef __MANIFEST_H__
 #define __MANIFEST_H__
 
-#include <cstdint>
 #include <list>
 #include "file_entry.h"
 
 namespace bundle
 {
     // Manifest Header contains:
-    // Fixed size thunk (represened by manifest_header_inner_t)
+    // Fixed size thunk (represened by manifest_header_fixed_t)
     //   - Major Version     
     //   - Minor Version     
     //   - Number of embedded files
-    //   - Bundle ID length 
     // Variable size portion:
+    //   - Bundle ID length 
     //   - Bundle ID ("Bundle ID length" bytes)
 
+#pragma pack(push, 1)
+    struct manifest_header_fixed_t
+    {
+        uint32_t major_version;
+        uint32_t minor_version;
+        int32_t num_embedded_files;
+    };
+#pragma pack(pop)
+
     struct manifest_header_t
     {
     public:
         manifest_header_t()
-            :m_data(), m_bundle_id()
+            :m_data(nullptr), m_bundle_id()
         {
         }
 
-        bool is_valid();
-        static manifest_header_t* read(FILE* stream);
-        const pal::string_t& bundle_id() { return m_bundle_id; }
-        int32_t num_embedded_files() { return m_data.num_embedded_files;  }
+        bool is_valid() const;
+        static manifest_header_t* read(reader_t &reader);
+        const pal::string_t& bundle_id() const { return m_bundle_id; }
+        int32_t num_embedded_files() const { return m_data->num_embedded_files;  }
 
     private:
-#pragma pack(push, 1)
-        struct
-        {
-            uint32_t major_version;
-            uint32_t minor_version;
-            int32_t num_embedded_files;
-            int8_t bundle_id_length_byte_1;
-        } m_data;
-#pragma pack(pop)
+        const manifest_header_fixed_t* m_data;
```

Comment:
In `file_entry` you make a copy of the memory mapped data to avoid IO later on. Here you don't. I don't know how big of a problem the IO can be, but we should be consistent. So either one or the other, but everywhere.

Reply:
As noted in the comments [here](https://github.com/dotnet/core-setup/pull/6576/files#diff-3d1ffec12bf784039f08581e68f7533fR41), the difference is in the usage pattern between the two kinds of headers. 

* I don't think we should read file-entry data from the bundle-file every time it is needed (ex: when loading an IL image in the middle of execution). It may cause additional file I/O. 

* The bundle header/footer on the other hand, is only read during startup. Of course, we can make a copy of the data here, but it is unnecessary, and will be discarded immediately.

The two are different structures with different lifetimes. So, I don't see why they need to be treated similarly.

Reply:
Makes sense - maybe just add a comment... thanks!